tfhe/src/core_crypto/commons/math/decomposition/decomposer.rs

```diff
@@ -219,15 +227,26 @@ where
         level_count: DecompositionLevelCount,
         ciphertext_modulus: CiphertextModulus<Scalar>,
     ) -> Self {
-        debug_assert!(
-            Scalar::BITS > base_log.0 * level_count.0,
-            "Decomposed bits exceeds the size of the integer to be decomposed"
+        assert!(
+            !ciphertext_modulus.is_power_of_two(),
+            "Got a power of 2 modulus as input for SignedDecomposerNonNative, \
+            this is not supported, use SignedDecomposer instead"
         );
-        Self {
+
+        let sself = Self {
```

Comment:
Typo? self instead of sself?

Reply:
self is a rust keyword, so when needed to use self we tend to call it sself

https://doc.rust-lang.org/reference/keywords.html :)